sdk/monitor/monitor-query/review/monitor-query.api.md

```diff
@@ -0,0 +1,293 @@
+};
+
+// @public
+export interface ErrorDetail {
+    additionalProperties?: any;
+    code: string;
+    message: string;
+    resources?: string[];
+    target?: string;
+    value?: string;
+}
+
+// @public
+export interface ErrorInfo {
+    additionalProperties?: any;
+    code: string;
+    details?: ErrorDetail[];
+    innererror?: ErrorInfo;
+    message: string;
+}
+
+// @public
+export interface GetMetricDefinitionsOptions extends OperationOptions {
+    metricNamespace?: string;
+}
+
+// @public
+export interface GetMetricDefinitionsResponse {
+    definitions: MetricDefinition[];
+}
+
+// @public
+export interface GetMetricNamespacesOptions {
+    startTime?: string;
+}
+
+// @public
+export interface GetMetricNamespacesResponse {
+    namespaces: MetricNamespace[];
+}
+
+// @public
+export interface LocalizableString {
+    localizedValue?: string;
+    value: string;
+}
+
+// @public
+export class LogsClient {
+    constructor(tokenCredential: TokenCredential, options?: LogsClientOptions);
+    // (undocumented)
+    queryLogs(workspaceId: string, query: string, options?: QueryLogsOptions): Promise<QueryLogsResult>;
+    // (undocumented)
+    queryLogsBatch(batch: QueryLogsBatch, options?: QueryLogsBatchOptions): Promise<QueryLogsBatchResponse>;
+}
+
+// @public (undocumented)
+export interface LogsClientOptions extends PipelineOptions {
+    host?: string;
+}
+
+// @public
+export interface MetadataValue {
+    name?: LocalizableString;
+    value?: string;
+}
+
+// @public
+export interface Metric {
+    displayDescription: string;
+    errorCode?: string;
+    id: string;
+    name: LocalizableString;
+    timeseries: TimeSeriesElement[];
+    type: string;
+    unit: MetricUnit;
+}
+
+// @public
+export interface MetricAvailability {
+    retention?: string;
+    timeGrain?: string;
+}
+
+// @public
+export interface MetricColumn {
+    name?: string;
+    type?: ColumnDataType;
+}
+
+// @public
+export interface MetricDefinition {
+    category?: string;
+    dimensions?: LocalizableString_2[];
+    displayDescription?: string;
+    id?: string;
+    isDimensionRequired?: boolean;
+    metricAvailabilities?: MetricAvailability[];
+    // Warning: (ae-forgotten-export) The symbol "LocalizableString" needs to be exported by the entry point index.d.ts
+    name?: LocalizableString_2;
+    primaryAggregationType?: AggregationType;
+    resourceId?: string;
+    // Warning: (ae-forgotten-export) The symbol "MetricUnit" needs to be exported by the entry point index.d.ts
+    unit?: MetricUnit_2;
+}
+
+// @public
+export interface MetricNamespace {
+    classification?: NamespaceClassification;
+    id?: string;
+    name?: string;
+    properties?: MetricNamespaceName;
+    type?: string;
+}
+
+// @public
+export interface MetricNamespaceName {
+    metricNamespaceName?: string;
+}
+
+// @public (undocumented)
+export class MetricsClient {
+    constructor(tokenCredential: TokenCredential, options?: MetricsClientOptions);
```

Comment:
I would expect `resourceUri` would be part of the constructor instead of being a required param on all methods? Or am I greatly confused on how this endpoint works?

Reply:
We can see how we make this more clear - it's not the arm resource id for the metrics server, it's the resource id that you're querying metrics on (ie, it could be a service bus, text analytics cluster, etc..). So it should be a parameter, but this is an oddity we'll want to consider.

Reply:
Captured this concern here: https://github.com/Azure/azure-sdk-for-js/issues/15296